qkernel/src/interrupt/x86_64/mod.rs

```diff
@@ -773,3 +792,111 @@ pub extern "C" fn TripleFaultHandler(sf: &mut PtRegs) {
     info!("\nTripleFaultHandler: at {:#x}\n{:#?}", sf.rip, sf);
     loop {}
 }
+
+#[cfg(feature = "cc")]
+#[no_mangle]
+pub extern "C" fn VmmCommunicationHandler(sf: &mut PtRegs, errorCode: u64) {
+    use crate::Kernel_cc::LOG_AVAILABLE;
+    let log_available = LOG_AVAILABLE.load(core::sync::atomic::Ordering::Acquire);
+    let from_user = sf.cs & 0x3 != 0;
+    if from_user {
+        CPULocal::Myself().SetMode(VcpuMode::Kernel);
+        let currTask = Task::Current();
+        let mut rflags = sf.eflags;
+        rflags &= !USER_FLAGS_CLEAR;
+        rflags |= USER_FLAGS_SET;
+        sf.eflags = rflags;
+        sf.ss |= 3;
+        currTask.SaveFp();
+    }
+    /* 
```

Comment:
Can you please remove this. And also any leftover commented code that you don't need. Also consider using `debug!()` or `error!()` depending on situation...